### PR TITLE
Revise list of Secret types

### PR DESCRIPTION
**Pull request by sftim**
This is a subset of the changes from PR #30984

I've split them into their own PR in the hope of seeing them merged even if the other PR is quite a lot of change to review.

This PR includes a fix for some inaccurate advice in the [Docker config Secrets](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets) section.

### CODE REVIEW

File: content/en/docs/concepts/configuration/secret.md [hunk 1]
```diff
@@ -211,33 +220,32 @@ to create a Docker registry Secret, you can do:
 ```shell
 kubectl create secret docker-registry secret-tiger-docker \
   --docker-username=tiger \
-  --docker-password=pass113 \
-  --docker-email=tiger@acme.com \
-  --docker-server=my-registry.example:5000
```
Comment (sftim) [inline content/en/docs/concepts/configuration/secret.md:216]: If it's important to include `--docker-server`, I can update the example to match it.

File: content/en/docs/concepts/configuration/secret.md [hunk 2]
```diff
@@ -95,11 +95,16 @@ Kubernetes imposes on them.
 | `bootstrap.kubernetes.io/token` | bootstrap token data |
 
 You can define and use your own Secret type by assigning a non-empty string as the
-`type` value for a Secret object. An empty string is treated as an `Opaque` type.
+`type` value for a Secret object (an empty string is treated as an `Opaque` type).
```
Comment (tengqm) [inline content/en/docs/concepts/configuration/secret.md:98]: Don't see a reason for this change

Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:98]: In spoken English, the speaker might shift tone slightly to indicate that the phrase “an empty string is treated as an Opaque type” is an associated details. In written English, the parens [are my attempt to] convey that.

File: content/en/docs/concepts/configuration/secret.md (hunk 1 above)
Comment (reylejano) [reply content/en/docs/concepts/configuration/secret.md:216]: Imo, I like keeping `--docker-password` , `--docker-email` , `--docker-server` as it's a comprehensive example.
They can also be found in the [pull images from a private registry task](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line)

**Comment by reylejano** [comment]
I have one small nit in an earlier [comment](https://github.com/kubernetes/website/pull/31042#discussion_r781645267) but overall lgtm
/lgtm

File: content/en/docs/concepts/configuration/secret.md (hunk 1 above)
Comment (sftim) [reply content/en/docs/concepts/configuration/secret.md:216]: I plan to update this.

**Comment by sftim** [comment]
I plan to fix this up
/lgtm cancel

**Pull request merged**